src/server/proxy/server/api_server.go

```diff
@@ -29,6 +29,17 @@ func (a *APIServer) Listen(request *proxy.ListenRequest, server proxy.API_Listen
 	if err := listener.Register(notifier); err != nil {
 		return errors.EnsureStack(err)
 	}
+
+	// send initial empty event to indicate to client that the listener has been registered
+	if err := server.Send(&proxy.ListenResponse{
+		Extra: "",
+	}); err != nil {
+		notifier.sendError(err)
+		return
```

Comment:
```suggestion
		return err
```
pretty sure this is all you need since we haven't started the notifier routine yet

Reply:
oh actually as is you wouldn't return the error, that's a weird pitfall of named return values

Reply:
eek! good catch